syntax = "proto2";
import "mapping_options.proto";

package tc;

message DLDataTypeProto {
  required uint32 code = 1;
  required uint32 bits = 2;
  required uint32 lanes = 3;
};

message TensorInfoProto {
  repeated uint64 shape = 1;
  repeated uint64 strides = 2;
  required uint64 alignment = 3;
  required DLDataTypeProto dtype = 4;
}

message OptionsCacheValuesProto{
  required CudaMappingOptionsProto kernel_options = 1;
  repeated uint64 recorded_runtimes = 2;
}

message OptionsCacheEntryProto {
  required string id = 1;
  repeated TensorInfoProto inputs = 2;
  repeated TensorInfoProto outputs = 3;
  required string device_str = 4;
  required string git_version = 5;

  repeated OptionsCacheValuesProto values = 6;
}

message OptionsCacheProto {
  repeated OptionsCacheEntryProto entries = 1;
}
